lib/task-graph.js

```diff
@@ -522,6 +524,35 @@ function taskGraphFactory(
         }));
     };
 
+    TaskGraph.prototype.updateGraphProgress = function(progressData) {
```

Comment:
This is a potential problem for this prototype function is that there is no reference for `this`.  Seems the input `progressData` has already included the graphId, but you don't check whether the input graphId is matched with this graph instance, so that to call this API, the progress of graph A may be wrongly updated into graph B.

You can either add some assertion logic to make sure the graph id is matched, or you can auto-fill the graphId for `progressData`, so that the `progressData` doesn't need contain the `graphId`.